akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/FilterStage.scala

```diff
@@ -297,7 +295,7 @@ import akka.stream.stage.StageLogging
             val env = grab(inEnv)
             val pid = env.persistenceId
 
-            if (filter.matches(pid)) {
+            if (filter.matches(pid) || envelopeFilter(env)) {
```

Comment:
maybe add a comment in above `onReplay` that it is intentionally not using the envelopeFilter there